### PR TITLE
Culling improvements + GRAPH_RENDER_INVISIBLE check change

### PR DESCRIPTION
**Pull request by Lilaa3**
GRAPH_RENDER_INVISIBLE is now checked before computing any transformation matrix for the object. Half fov for vertical and horizontal culling is now computed during geo_process_perspective. Vertical culling has been added.
Emulators have basically infinite culling aspect ratio to prevent early culling with widescreen viewport hacks. This was written by both me and Kaze Emanuar, he provided the suggestion to use absolute values to reduce conditions.

### CODE REVIEW

File: src/engine/graph_node.h [hunk 1]
```diff
@@ -132,9 +132,11 @@ struct GraphNodeOrthoProjection {
 struct GraphNodePerspective {
     /*0x00*/ struct FnGraphNode fnNode;
     /*0x18*/ s32 unused;
-    /*0x1C*/ f32 fov;   // horizontal field of view in degrees
-    /*0x20*/ u16 near;  // near clipping plane
-    /*0x22*/ u16 far;   // far clipping plane
+    /*0x1C*/ f32 halfFovHorizontal;
+    /*0x20*/ f32 halfFovVertical;
+    /*0x24*/ f32 fov;   // horizontal field of view in degrees
+    /*0x26*/ u16 near;  // near clipping plane
```
Comment (gheskett) [inline src/engine/graph_node.h:138]: This offset is not correct. Also is there actually any reason for us to leave the `unused` field in this struct?

File: src/game/rendering_graph_node.c [hunk 2]
```diff
@@ -1022,7 +1044,7 @@ void geo_process_shadow(struct GraphNodeShadow *node) {
 /**
  * Check whether an object is in view to determine whether it should be drawn.
  * This is known as frustum culling.
- * It checks whether the object is far away, very close / behind the camera,
+ * It checks whether the object is far away, very close / behind the camera, vertically,
  * or horizontally out of view. It does not check whether it is vertically
```
Comment (gheskett) [inline src/game/rendering_graph_node.c:1048]: Description contradicts itself

File: include/config/config_graphics.h [hunk 3]
```diff
@@ -133,3 +133,20 @@
...
+/**
+ * Makes objects bellow the screen be culled.
+*/
+#define VERTICAL_CULLING
+
+/**
+ * If the first command of an object´s geolayout is not GEO_CULLING_RADIUS, DEFAULT_CULLING_RADIUS
+ * will be used instead.
+ */
+#define DEFAULT_CULLING_RADIUS 300
```
Comment (gheskett) [inline include/config/config_graphics.h:152]: Nitpick, but please add the newline here lmao

**Review by gheskett — CHANGES_REQUESTED** [review]

**Review by gheskett — APPROVED** [review]
Additional review needed before merging

File: src/game/rendering_graph_node.c [hunk 4]
```diff
@@ -1051,54 +1074,46 @@ void geo_process_shadow(struct GraphNodeShadow *node) {
...
+
+#endif
+    
+    f32 hScreenEdge = -cameraToObjectDepth * gCurGraphNodeCamFrustum->halfFovHorizontal;
+
+    if (absf(node->cameraToObject[0]) > hScreenEdge + cullingRadius) {
         return FALSE;
     }
-
+
```
Comment (thecozies) [inline src/game/rendering_graph_node.c:1116]: some extra whitespace is here

```suggestion

```

**Review by thecozies — APPROVED** [review]
super minor changes but i tested functionality and it works really well!! this is a nice and cleanly written optimization

approving but please add those suggestions i dropped here, then we can merrrrgggggeeeee

**Pull request merged**